File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        char_count = {}
        for c in s:
            char_count[c] = char_count.get(c, 0) + 1
        for c in t:
            if c in char_count:
                char_count[c] -= 1
                if char_count[c] == 0:
                    char_count.pop(c)
            else:
                return False
        return len(char_count) == 0

File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length()) return false;
        unordered_map<char, int> char_map {};
        for (char c : s) {
            if (char_map.find(c)!=char_map.end()) {
                char_map[c]++;
            }
            else {
                char_map[c] = 1;
            }
        }
        for (char c : t) {
            if (char_map.find(c)!=char_map.end()) {
                char_map[c]--;
                if (char_map[c] == 0) {
                    char_map.erase(c);
                }
            }
            else return false;
        }
        return true;
    }
};